client/src/context/DataContext.jsx

```diff
@@ -11,11 +11,12 @@ export const DataProvider = ({ children }) => {
   const baseApiUrl = process.env.REACT_APP_API_URL;
   let endpoint = "";
 
-  switch (currentPage) {
-    case "profile":
-      endpoint = "/api/get-workouts-for-user";
+  switch (true) {
```

Comment:
is this meant to be left as hardcoded true?

Reply:
yes, the syntax of this statement is meant to be run with the cases below
`switch (true)
     case currentPage.startsWith("profile/"):
     code logic`
this basically means: 
`if currentPage starts with "profile/", then execute the code logic in the case block`